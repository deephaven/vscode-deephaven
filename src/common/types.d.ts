export type ConnectionType = 'DHC';

export type ConnectionAndSession<TConnection, TSession> = {
  cn: TConnection;
  session: TSession;
};
