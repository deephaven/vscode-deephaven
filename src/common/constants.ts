export const CONFIG_KEY = 'vscode-deephaven';
export const CONFIG_CORE_SERVERS = 'core-servers';

// export const DHFS_SCHEME = 'dhfs';
export const RUN_CODE_COMMAND = `${CONFIG_KEY}.runCode`;
export const RUN_SELECTION_COMMAND = `${CONFIG_KEY}.runSelection`;
export const SELECT_CONNECTION_COMMAND = `${CONFIG_KEY}.selectConnection`;

export const STATUS_BAR_DISCONNECTED_TEXT = 'Deephaven: Disconnected';
export const STATUS_BAR_DISCONNECT_TEXT = 'Deephaven: Disconnect';
