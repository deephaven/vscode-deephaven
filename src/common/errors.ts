export class QueryCreationCancelledError extends Error {
  constructor() {
    super('Query creation cancelled');
    this.name = 'QueryCreationCancelledError';
  }
}

export class UnsupportedConsoleTypeError extends Error {
  constructor(message: string) {
    super(message);
    this.name = 'UnsupportedConsoleTypeError';
  }
}
