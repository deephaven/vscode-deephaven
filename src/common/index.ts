export * from './constants';
export type * from './types';
