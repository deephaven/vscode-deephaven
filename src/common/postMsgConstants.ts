/**
 * Constants used for `postMessage` interaction.
 * Note that these need to be consumable from both the extension code (CJS) and
 * the webview content code (ESM), so avoid importing anything from other files
 * to minimize the risk of breaking the builds.
 */

export const DEEPHAVEN_POST_MSG = {
  authTokenRequest: 'io.deephaven.message.LoginOptions.authTokenRequest',
  loginOptionsRequest: 'io.deephaven.message.LoginOptions.request',
  sessionDetailsRequest: 'io.deephaven.message.SessionDetails.request',
  workerCreated: 'io.deephaven.message.LoginOptions.workerCreated',
} as const;

export const VSCODE_POST_MSG = {
  loginOptionsResponse: 'vscode-ext.loginOptions',
  sessionDetailsResponse: 'vscode-ext.sessionDetails',
} as const;
