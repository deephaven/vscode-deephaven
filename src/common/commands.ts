import { EXTENSION_ID } from './constants';

export const CONNECT_TO_SERVER_CMD = `${EXTENSION_ID}.connectToServer`;
export const CREATE_NEW_TEXT_DOC_CMD = `${EXTENSION_ID}.createNewTextDoc`;
export const DISCONNECT_EDITOR_CMD = `${EXTENSION_ID}.disconnectEditor`;
export const DISCONNECT_FROM_SERVER_CMD = `${EXTENSION_ID}.disconnectFromServer`;
export const DOWNLOAD_LOGS_CMD = `${EXTENSION_ID}.downloadLogs`;
export const OPEN_IN_BROWSER_CMD = `${EXTENSION_ID}.openInBrowser`;
export const REFRESH_SERVER_TREE_CMD = `${EXTENSION_ID}.refreshServerTree`;
export const REFRESH_SERVER_CONNECTION_TREE_CMD = `${EXTENSION_ID}.refreshServerConnectionTree`;
export const RUN_CODE_COMMAND = `${EXTENSION_ID}.runCode`;
export const RUN_SELECTION_COMMAND = `${EXTENSION_ID}.runSelection`;
export const SELECT_CONNECTION_COMMAND = `${EXTENSION_ID}.selectConnection`;
export const START_SERVER_CMD = `${EXTENSION_ID}.startServer`;
export const STOP_SERVER_CMD = `${EXTENSION_ID}.stopServer`;
