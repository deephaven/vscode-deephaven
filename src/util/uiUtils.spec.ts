import { describe, it, expect, vi } from 'vitest';
import {
  createConnectTextAndTooltip,
  ConnectionOption,
  createConnectionOptions,
} from './uiUtils';
import { ConnectionConfig } from '../common';

// See __mocks__/vscode.ts for the mock implementation
vi.mock('vscode');

describe('createConnectionOptions', () => {
  const configs: ConnectionConfig[] = [
    { url: 'http://localhost:10000', consoleType: 'python' },
    { url: 'http://localhost:10001', consoleType: 'groovy' },
  ];

  it.each(configs)(
    'should return connection options: $url:$consoleType',
    config => {
      const actual = createConnectionOptions([config]);
      expect(actual).toMatchSnapshot();
    }
  );
});

describe('createConnectTextAndTooltip', () => {
  const option: ConnectionOption = {
    type: 'DHC',
    consoleType: 'python',
    label: 'DHC: localhost:10000',
    url: 'http://localhost:10000',
  };

  it.each([['connecting'], ['connected'], ['disconnected']] as const)(
    `should return text and tooltip: '%s'`,
    status => {
      const actual = createConnectTextAndTooltip(status, option);
      expect(actual).toMatchSnapshot();
    }
  );
});
