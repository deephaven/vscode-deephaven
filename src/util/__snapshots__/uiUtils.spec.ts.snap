// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`createConnectTextAndTooltip > should return text and tooltip: 'connected' 1`] = `
{
  "text": "$(vm-connect) DHC: localhost:10000",
  "tooltip": "Connected to http://localhost:10000",
}
`;

exports[`createConnectTextAndTooltip > should return text and tooltip: 'connecting' 1`] = `
{
  "text": "$(sync~spin) Deephaven: Connecting...",
  "tooltip": "Connecting to http://localhost:10000...",
}
`;

exports[`createConnectTextAndTooltip > should return text and tooltip: 'disconnected' 1`] = `
{
  "text": "$(debug-disconnect) Deephaven: Disconnected",
  "tooltip": "Connect to Deephaven",
}
`;

exports[`createConnectionOptions > should return connection options: 'http://localhost:10000':'python' 1`] = `
[
  {
    "consoleType": "python",
    "label": "DHC: localhost:10000",
    "type": "DHC",
    "url": "http://localhost:10000",
  },
]
`;

exports[`createConnectionOptions > should return connection options: 'http://localhost:10001':'groovy' 1`] = `
[
  {
    "consoleType": "groovy",
    "label": "DHC: localhost:10001",
    "type": "DHC",
    "url": "http://localhost:10001",
  },
]
`;
